sql/files/defaultdata/compare/compare.cc

```diff
@@ -72,30 +73,45 @@ FILE *openfeedback(const char *feedbackdir, const char *feedback, const char *wh
 	return res;
 }
 
+// The behavior of std::tolower on (signed) char is undefined.
+char tolower_char(char c)
+{
+    return static_cast<char>(std::tolower(static_cast<unsigned char>(c)));
+}
+
+bool equal_case_insensitive(std::string a, std::string b)
+{
+	for (size_t i=1; i<a.length(); i++) a[i] = tolower_char(a[i]);
+	for (size_t i=1; i<b.length(); i++) b[i] = tolower_char(b[i]);
```

Comment:
Why do these loops start at `1`, and not at `0`? If I now print `Impossible` instead of `impossible`, this is not allowed according to this validator, even though it should be case-insensitive :sweat_smile: 

(big thanks to `NickyAndMark.java` at problem I of BAPC 2022 for triggering this :stuck_out_tongue:)

Reply:
These Nicky and Mark are great guys